Sources/Runes.swift

```diff
@@ -1,93 +1,75 @@
+precedencegroup AlternativePrecedence {
+    associativity: left
+    lowerThan: ComparisonPrecedence
+}
+
+precedencegroup MonadicPrecedenceRight {
+    associativity: right
+    lowerThan: LogicalDisjunctionPrecedence
+    higherThan: AssignmentPrecedence
+}
+
+precedencegroup MonadicPrecedenceLeft {
+    associativity: left
+    lowerThan: LogicalDisjunctionPrecedence
+    higherThan: AssignmentPrecedence
+}
+
 /**
 map a function over a value with context
 
 Expected function type: `(a -> b) -> f a -> f b`
 Haskell `infixl 4`
 */
-infix operator <^> {
-    associativity left
-
-    // Same precedence as the equality operator (`==`)
-    precedence 130
-}
+infix operator <^>: ComparisonPrecedence
 
 /**
 apply a function with context to a value with context
 
 Expected function type: `f (a -> b) -> f a -> f b`
 Haskell `infixl 4`
 */
-infix operator <*> {
-    associativity left
-
-    // Same precedence as the equality operator (`==`)
-    precedence 130
-}
+infix operator <*>: ComparisonPrecedence
 
 /**
 sequence actions, discarding right (value of the second argument)
 
 Expected function type: `f a -> f b -> f a`
 Haskell `infixl 4`
 */
-infix operator <* {
-    associativity left
-    precedence 140
-}
+infix operator <*: ComparisonPrecedence
 
 /**
 sequence actions, discarding left (value of the first argument)
 
 Expected function type: `f a -> f b -> f b`
 Haskell `infixl 4`
 */
-infix operator *> {
-    associativity left
-    precedence 140
-}
+infix operator *>: ComparisonPrecedence
 
 /**
 an associative binary operation
 
 Expected function type: `f a -> f a -> f a`
 Haskell `infixl 3`
 */
-infix operator <|> {
-    associativity left
-
-    // Lower precedence than `<^>`, `<*>`, `*>`, `<*`
-    precedence 120
-}
+infix operator <|>: AlternativePrecedence
```

Comment:
Yeah, fair point. This operator is from the `Alternative` typeclass in Haskell, which is why I named it as such. I'm totally open to dropping that baggage though.